tiledb/sm/group/group.cc

```diff
@@ -300,6 +300,50 @@ Status Group::get_query_type(QueryType* query_type) const {
   return Status::Ok();
 }
 
+void Group::delete_group(const URI& uri, bool recursive) {
+  // Check that group is open
+  if (!is_open_) {
+    throw StatusException(
```

Comment:
Declare a class for these in group.cc... See DeleteAndUpdateStatusException for an example.